client/lib/i18n-utils/switch-locale.js

```diff
@@ -547,15 +547,33 @@ function loadCSS( cssUrl, currentLink ) {
 }
 
 /**
- * Add translations.
+ * Translation data batch strore.
+ *
+ * @type {Array}
+ */
+const _translationsBatch = [];
+
+/**
+ * A throttle wrapper around i18n.addTranslations.
+ *
+ * @param {Object} userTranslations User translations data that will override chunk translations
+ */
+const _addTranslationsBatch = throttle( function ( userTranslations ) {
+	i18n.addTranslations( Object.assign( {}, ..._translationsBatch.splice( 0 ), userTranslations ) );
+}, 100 );
+
+/**
+ * Adds new translations to the existing locale data.
  *
  * This function also saves the duration of the call as a performance measure
  *
- * @param translations translations to add
+ * @param {Object} translations       Translations data
+ * @param {Object} [userTranslations] User translations data that will override chunk translations
  */
-function addTranslations( translations ) {
+function addTranslations( translations, userTranslations ) {
 	window.performance?.mark( 'add_translations_start' );
```

Comment:
Shouldn't this metrics be inside the throttled function?

Reply:
I had them initially in there, but then realized that it won't be reporting `translationsChunksCount` correctly, so I moved it outside.

Reply:
I don't think this is measuring it correctly to be honest.
For `translationsChunksCount` I'd expect the number to go down when we deploy the throttling. We can use that as an indicative of throttling working.
For `translationsChunksDuration` I'd expect it to measure the actual real calls to `i18n.addTranslations`. The fact that Calypso "aggregates" some of those calls via throttling shouldn't be reflected in the numbers.

Reply:
Good point, I'll move it back.